### ISSUE TITLE
[iOS Bug] IndexOutOfRange when setting the Text property

### ISSUE THREAD

**Issue by Phenek** [xabre/xamarin-forms-tab-badge#62]
My implementation is like so: 
```
            var tab = new AccountPage();
            var navigation = new NavigationPage(tab)
            {
                Title = "Profil",
                Icon = "Profile",
            };

            tab.SetBinding(TabBadge.BadgeTextProperty, nameof(BottomMenuViewModel.Count));
            tab.SetBinding(TabBadge.BadgeColorProperty, nameof(BottomMenuViewModel.Color));
            tab.SetBinding(TabBadge.BadgeTextColorProperty, nameof(BottomMenuViewModel.TextColor));
            tab.BindingContext = viewModel;
            Children.Add(navigation);
```

When changing the BadgeTextProperty I got an IndexOutOfRange on iOS Only.
Message: Index was outside the bounds of the array.

The StackTrace take me to the Plugin.Badge.iOS.BadgedTabbedPageRenderer.OnTabbedPagePropertyChanged

**Comment by Phenek** [comment]
I think is about the CheckValidTabIndex function in [the iOS Renderer](https://github.com/xabre/xamarin-forms-tab-badge/blob/master/Source/Plugin.Badge.iOS/BadgedTabbedPageRenderer.cs)

The function is searching for the IndexOf(tab) and need to be the IndexOf(navigation)
regarding my implementation. #58

**Comment by Phenek** [comment]
For a workaround we could do : 
```
        public bool CheckValidTabIndex(Page page, out int tabIndex)
        {
            tabIndex = Tabbed.Children.IndexOf(page);
            if(tabIndex == -1 && page.Parent != null)
            {
                 tabIndex = Tabbed.Children.IndexOf(page.Parent);
            }
            return tabIndex < TabBar.Items.Length && tabIndex > 0;;
        }
```

**Comment by xabre** [comment]
PR #63 was merged and released with v2.1.1
